satpy/readers/fci_l2_nc.py

```diff
@@ -203,26 +206,36 @@ def get_area_def(self, key):
         """Return the area definition (common to all data in product)."""
         return self._area_def
 
-    def _compute_area_def(self):
+    def _compute_area_def(self, dataset_id):
         """Compute the area definition.
 
         Returns:
             AreaDefinition: A pyresample AreaDefinition object containing the area definition.
 
         """
-        # Read the projection data from the mtg_geos_projection variable
-        a = float(self._projection.attrs['semi_major_axis'])
-        b = float(self._projection.attrs['semi_minor_axis'])
-        h = float(self._projection.attrs['perspective_point_height'])
+        area_extent = self._get_area_extent()
+        area_naming, proj_dict = self._get_proj_area(dataset_id)
+        area_def = geometry.AreaDefinition(
+            area_naming['area_id'],
+            area_naming['description'],
+            "",
+            proj_dict,
+            self.ncols,
+            self.nlines,
+            area_extent)
+
+        return area_def
 
-        # TODO sweep_angle_axis value not handled at the moment, therefore commented out
+    def _get_area_extent(self):
+        """Calculate area extent of dataset."""
+        # TODO Sweep_angle_axis value not handled at the moment, therefore commented out
         # sweep_axis = self._projection.attrs['sweep_angle_axis']
 
         # Coordinates of the pixel in radians
         x = self.nc['x']
         y = self.nc['y']
-        # TODO conversion to radians: offset and scale factor are missing from some test NetCDF file
-        # TODO the next two lines should be removed when the offset and scale factor are correctly configured
+        # TODO Conversion to radians: offset and scale factor are missing from some test NetCDF file
+        # TODO The next three lines should be removed when the offset and scale factor are correctly configured
```

Comment:
Could all these TODOs be gathered in an issue on github? That way it's easier to track probably?

Reply:
Thanks for the review! I'm quite new to this reader, but I think a lot has happened since the previous version was written and it could be that some TODOs can be resolved/fixed now and then it's probably better indeed to fix those before merging. I will discuss internally and then let you know! I also agree that remaining TODOs can me moved to one or several issues instead.

Reply:
I have finally worked my way trough the reader and managed to resolve all TODOs, since most of them were either outdated or could be resolved with improved implementations. While I was at it I also added some additional datasets and features.